exosphere-shared/src/compile-service-routes.ls

```diff
@@ -17,4 +21,31 @@ module.exports = (app-config, base-path) ->
             sends: service-config.messages.sends
             namespace: service-data.namespace
           }
-  service-routes
+
+  # compile list of external service config
+  external-services = []
+  for protection-level of app-config.services
+    for role, service-data of app-config.services[protection-level]
+      if service-data['docker-image']
+        external-services.push {image: service-data['docker-image'], file-name: 'service.yml'}
+        
+  # compile service.yml in each external Docker containers
+  async.map-series external-services, (-> DockerHelper.cat-file &0, &1), (err, external-service-configs) ->
+    | err => return done null 
+
+    # compile external service routes
+    external-service-index = 0
+    for protection-level of app-config.services
+      for role, service-data of app-config.services[protection-level]
```

Comment:
Thoughts on adding the role and service data into `external-services` above and that way you don't need to loop through the app-config again?